the-collab-lab/tcl-27-smart-shopping-list
# Sk yh validations and calendar

## Description
* Implemented validations  for the itemName input and frequency radio buttons
* Imported react-datepicker to utilize a popup calendar for selecting a purchase date

## Related Issue
closes #22

## Acceptance Criteria
https://github.com/the-collab-lab/tcl-27-smart-shopping-list/issues/22

AC:
* Warn user when an item isn’t entered or a frequency isn’t selected
* Use react-datepicker to display a calendar when a user wants to pick a date for last purchased
* Calendar dates should be sent to the backend as a consistent date format

_Note: This issue is for extra implied features the team wanted to implement_

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria
* From your terminal, pull from main branch with `git pull origin main` 
* Pull this branch with `git pull origin sk-yh-validations-and-calendar` 
* Checkout branch with `git checkout sk-yh-validations-and-calendar`
* Update dependencies with npm update.

Submit a blank form and make sure you get a warning that an item must be entered and/or a frequency must be checked. Also, click on the Last Purchase input field and make sure a calendar popup appears. Click a date and check the List to see that the item has been rendered. You can also choose to clear the Last Purchase input field by clicking on the ‘x’ after clicking a date.
